internal/utility/vulns/vulnerability.go

```diff
@@ -125,9 +125,23 @@ func AffectsEcosystem(v osvschema.Vulnerability, ecosystemAffected osvecosystem.
 	return false
 }
 
+// NormalizeRepo applies some reasonable transformations to repository urls to
+// ensure accurate results when determining if two repository urls are referencing
+// the same repository.
+//
+// Specifically, common protocols are removed from the start of the url and the
+// ".git" suffix if present
+func NormalizeRepo(repo string) string {
```

Comment:
OSV.dev is a bit more aggressive when normalizing (removing *all* schemes, and trimming any possible trailing slashes) but I'm happy to do it more explicitly like you've done here.

I don't think there's any functional difference - I doubt we have any non-http(s)/git URLs in OSV.